libraries/chain/controller.cpp

```diff
@@ -330,8 +330,11 @@ struct controller_impl {
          // todo: move this check to startup so id does not have to be calculated
          EOS_ASSERT( root_id == log_head->calculate_id(), fork_database_exception, "fork database root does not match block log head" );
       } else {
-         EOS_ASSERT( fork_db.root()->block_num == lib_num, fork_database_exception,
-                     "empty block log expects the first appended block to build off a block that is not the fork database root" );
+         if( fork_db.root()->block_num != lib_num ) {
+            ilog( "Empty block log expects the first appended block to build off a block that is not the fork database root, terminating. root block number: ${block_num}, lib: ${lib_num}", ("block_num", fork_db.root()->block_num) ("lib_num", lib_num) );
+            shutdown();
```

Comment:
This introduces unique error handling behavior for this one case. I think it would be better to handle the fork_database_exception differently in the producer_plugin.

Reply:
Was worried exit on any fork_database_exception would impact wider areas. But once fork_db becomes corrupt, nothing can be done. Code changed according to the comments. Thanks.